prisma/migrations/20230804135730_initial_state/migration.sql

```diff
@@ -1,25 +0,0 @@
--- CreateTable
```

Comment:
Dúvida:

Se vc tivesse amanhã que alterar o schema de alguma tabela em produção.

Como vc faria esse SQL Migration usando o PG?

Reply:
Eu nunca enfrentei uma migração em produção.. Não entendo muito bem os B.Os. Eu teria que fazer os ''ALTER TABLES'', imagino. Vou pesquisar sobre o assunto.

Reply:
Pra mim é bastante nebuloso: "Os problemas de produção"... Nunca coloquei algum codigo em produção então sempre fico: "??" 
Mas sempre fico em duvida, tipo "caso a alteração na estrutura do banco de dados seja grande demais: o que fazer com os dados? " 

Se esse task caisse na minha mão, em nível "produção" sem backup (que eu não sei fazer), eu confesso que seguiria os seguintes passos:

1.  Subir outro banco de dados seguindo a estrutura mais atual;
2. Criaria um codigo intermediario cuja função é "ADAPTAR" os dados do banco de dados antigo para o novo.
3. Rodaria esse script e validaria de alguma forma (ainda desconheço como) que todos os dados foram transferidos.
4. Subiria o novo sistema (que lida com o novo banco de dados) enquanto removo aos poucos o banco de dados antigo e os serviços antigos.

O que voce acha? haha Sim, parece ineficiente caso o banco de dados seja enorme hUEHEU, não faço ideia quanto tempo isso levaria da mesma forma, ou, qual a ordem desse algoritmo.
:V

Reply:
É definitivamente uma alternativa.

SQL Migration é um rolê meio cabeludo msm.

Eu perguntei pq o prisma já te oferece uma forma de executar "SQL Migration". Usando PG, isso meio que fico sob sua responsabilidade :B 

Mas sendo honesto, eu tive q fazer poucas vezes SQL Migration.

Pode fechar essa thread qd vc ler isso, mas fica dá uma lida depois:
- O que é SQL Migration. Qual problema resolve? Como geralmente é feito?

Reply:
Mano, depende do migration, o prisma falha também.. Tipo, eu fiz algumas migrações (para seguir procedimento) e em uma delas alterei parametro UNIQUE (não me lembro ao certo o detalhes). 
Nessa migração em específico, o NPX do prisma reclamou que não conseguiria migrar os dados e **portanto precisaria limpar todo o banco de dados**

To dizendo só para trocar figurinha mesmo... Pode ser que exista alguma alternativa intermediaria que não gere esse problema.

Reply:
Não, mas é isso msm, vc tem razão:

SQL Migration n é perfeito, e é dificil.

O mais importante pra mim é vc entender isso:
> O que é SQL Migration? Qual problema resolve? Como geralmente é feito?

E sacar q uma possível "vantagem" desses ORMs, é q ele facilita pra vc fazer SQL Migration (facilita msm? xD)

Reply:
Pra brincar de "SQL Migration" n precisa estar em produção.

Qlq DB local q tenha dados q vc não queira perder, serve pra brincar de SQL Migration :V